### DISCUSSION TITLE
Klipper error triggers print cannceld

### DISCUSSION THREAD

**Discussion by Gagonjh** [OctoFarm/OctoFarm#416] [General]
My second printer have an issue with the pid values that triggers the firmware error. Each time a warning comes from klipper it seems the server counts this as an cancelled print. 

![OCtofarm_History_Klipper_2](https://user-images.githubusercontent.com/49719775/113514914-84fdca00-9571-11eb-8a53-14e2eb75d21a.PNG)
![OCtofarm_History_Klipper](https://user-images.githubusercontent.com/49719775/113514912-829b7000-9571-11eb-9fc3-8d00644d4872.PNG)

[Logs.zip](https://github.com/OctoFarm/OctoFarm/files/6254908/Logs.zip)

**Comment by NotExpectedYet** [comment]
Where was the klipper issue you mentioned?

**Reply by NotExpectedYet** [reply]
I believe this is due to the fact that we disable the disconnect when adding klipper into the mix with OctoPrint. 

Specifically this setting: 
`Cancel any ongoing prints but stay connected to the printer Please note that if you choose this, OctoPrint will still disconnect from the printer in case of fatal errors reported by your firmware (e.g. kill() called, fatal:).`

I wouldn't really know what to do, as with the connection still established OctoPrint will keep reporting those errors through the websocket. 

Turning that off and cancelling would put klipper in a dirty state so it isn't advised as it would cause reconnection issues without running a firmware_restart, or could cause other issues with Klippers error reporting.

**Reply by Gagonjh** [reply]
Yep the problem is between Octoprint and Klipper. I spotted M105s or M108s causing an repeated error.
 
The problem with the M105 is that every time the answer is "T:0" Octoprint immediately sends another M105.
![Screenshot 2021-04-05 152636](https://user-images.githubusercontent.com/49719775/113578708-85ef3400-9623-11eb-8b8e-a8e8ed2db8f0.jpg)
![Screenshot 2021-04-05 152705](https://user-images.githubusercontent.com/49719775/113578718-88ea2480-9623-11eb-85fd-3d5a9a8cd535.jpg)

For the M108 I don't know why it happened. But this code isn't used in Klipper so I just make an empty Script for it and hope for the best.

**Reply by NotExpectedYet** [reply]
> Yep the problem is between Octoprint and Klipper. I spotted M105s or M108s causing an repeated error.
> 
> The problem with the M105 is that every time the answer is "T:0" Octoprint immediately sends another M105. ![Screenshot 2021-04-05 152636](https://user-images.githubusercontent.com/49719775/113578708-85ef3400-9623-11eb-8b8e-a8e8ed2db8f0.jpg) ![Screenshot 2021-04-05 152705](https://user-images.githubusercontent.com/49719775/113578718-88ea2480-9623-11eb-85fd-3d5a9a8cd535.jpg)
> 
> For the M108 I don't know why it happened. But this code isn't used in Klipper so I just make an empty Script for it and hope for the best.

I believe there may be a fix coming for this. https://github.com/OctoPrint/OctoPrint/issues/4299
